company-box-icons.el

```diff
@@ -226,5 +226,12 @@ specification.md#completion-request-leftwards_arrow_with_hook.")
   (when (get-text-property 0 'yas-annotation candidate)
     'Template))
 
+(defun company-box-icons--tern (candidate)
+  "Return the icon type that corresponds to CANDIDATE type."
+  (when (derived-mode-p 'js2-mode)
```

Comment:
This is wrong. `company-tern` doesn't have to be used with `js2-mode`. You should test for the minor mode itself, not the major mode.

Reply:
I think both should be tested, first the major mode (and it probably has to be js-mode instead of js2.mode) and then do a (require company-tern nil t) just as a13 says, how about that?